### PR TITLE
Extend client API with custom HTTP requests

### PR DESCRIPTION
**Pull request by mat007**
**- What I did**

I made the client underlying HTTP client accessible, useful for instance for https://github.com/docker/cli/pull/1034

**- How I did it**

Added a `HTTPClient()` accessor to the `Client` interface.

**- A picture of a cute animal (not mandatory but encouraged)**

image

### CODE REVIEW

**Comment by cpuguy83** [comment]
I don't really see the context of why this would be needed in the mentioned issue.

Also would it be better to add a method to get the HTTP client rather than having the API client itself perform the request? It seems weird to have the API client just do any random request.

**Comment by mat007** [comment]
> I don't really see the context of why this would be needed in the mentioned issue.

It might be a bit hard to follow with much of the context being lost due to the many rebases.
Basically this was suggested by @vdemeester to be able to rewrite https://github.com/docker/cli/pull/1034/commits/0bf781e9123ae4ba287f183ab5a98225dc760950 into https://github.com/docker/cli/pull/1034/commits/05bbb2efc8837f15c674ac1a37fd0c7cbab17330

> Also would it be better to add a method to get the HTTP client rather than having the API client itself perform the request? It seems weird to have the API client just do any random request.

I really don't have enough experience with the design of moby and docker/CLI to answer that.
However the initial need was indeed a way to get access to the underlying HTTP client in order to use its configuration (mainly credentials), so I'd be happy with a good old `GetHTTPClient()`.
@vdemeester any though on this?

**Comment by vdemeester** [comment]
@mat007 yeah having a exported method to get the client's http client would work too :angel:

**Comment by mat007** [comment]
@cpuguy83 I changed the PR to now expose a `HTTPClient` getter instead.

File: client/client.go [hunk 1]
```diff
@@ -356,6 +356,11 @@ func (cli *Client) DaemonHost() string {
 	return cli.host
 }
 
+// HTTPClient returns the HTTP client bound to the server
+func (cli *Client) HTTPClient() *http.Client {
+	return cli.client
```
[the remark this answers is not in the document]
Comment (mat007) [reply client/client.go]: OK, I changed that. Still returning a pointer though as `http.Client` functions take a pointer as receptor and it's more convenient to directly get a pointer there.

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by thaJeztah — APPROVED** [review]
LGTM
